dffml/service/dev.py

```diff
@@ -1039,6 +1040,135 @@ async def run(self):
                 )
 
 
+class CommitLintError(Exception):
+    pass
+
+
+class CommitLint(CMD):
+    """
+    Enforce commit message style 
+    """
+
+    def _get_ignore_filter(self):
+        return lambda x: not x.endswith("_")
+
+    def _execute_func_list(self, param, func_list):
+        for func in func_list:
+            param = func(param)
+        return param
+
+    async def _get_file_mutations(self):
+        no_mutation = lambda x: x
+        mutation_func_factory = lambda ext: lambda x: x + ext
+        extensions = await self._get_all_exts()
+        mutations = {
+            "prefix_mutations": [
+                no_mutation,
+                lambda x: "." + x,
+                lambda x: "dffml/" + x,
+            ],
+            "suffix_mutations": [
+                no_mutation,
+                *[mutation_func_factory(ext) for ext in extensions],
+            ],
+        }
+        return mutations
+
+    async def _get_cmd_output(self, cmd: List[str]):
+        proc = await asyncio.create_subprocess_exec(
+            *cmd, stdout=asyncio.subprocess.PIPE
+        )
+        stdout, _ = await proc.communicate()
+        if proc.returncode != 0:
+            raise RuntimeError
+        output = stdout.decode().strip()
+        return output
+
+    async def _get_relevant_commits(self):
+        cmd = [
+            "git",
+            "log",
+            "--no-merges",
+            "--oneline",
+            "--format=%s",
+            await self._get_current_branch(),
+            "^master",  #! This needs to change when master is renamed to main.
+        ]
+        commits = await self._get_cmd_output(cmd)
+        commits_list = commits.split("\n")
+        return commits_list
+
+    async def _get_current_branch(self):
+        cmd = ["git", "branch", "--show-current"]
+        current_branch = await self._get_cmd_output(cmd)
+        return current_branch
+
+    async def _get_commmit_details(self, msg):
+        cmd = ["git", "log", f"--grep={msg}"]
+        commit_details = await self._get_cmd_output(cmd)
+        return commit_details
+
+    async def _get_all_exts(self):
+        cmd_1 = [
```

Comment:
```suggestion
        cmd = [
```

Reply:
Actually I was to pipe output of this command into another one but later on removed that part , which lead to this name , anyways changed :)